### PR TITLE
Add documentation for String#inspect and #dump methods

### PR DESCRIPTION
**Pull request by straight-shoota**
Also includes some minor code improvements.

### CODE REVIEW

File: src/string.cr [hunk 1]
```diff
@@ -3829,58 +3829,90 @@ class String
...
   end
 
-  def inspect(io)
+  # Pretty prints `self` into the given printer.
+  def pretty_print(pp : PrettyPrint) : Nil
+    pp.text(inspect)
+  end
+
+  # Returns a representation of `self` using character escapes for special characters and wrapped in quotes.
+  # ```crystal
```
Comment (Sija) [inline src/string.cr]: Please keep blank line before code example for readability.
Also, `crystal` here is not needed, code markers are enough.

File: src/string.cr [hunk 2]
```diff
@@ -3829,58 +3829,90 @@ class String
...
   end
 
-  def pretty_print(pp)
-    pp.text(inspect)
-  end
-
-  def inspect_unquoted
+  # Returns a representation of `self` using character escapes for special characters but not wrapped in quotes.
+  #
+  # ```crystal
```
Comment (Sija) [inline src/string.cr]: ditto

**Comment by straight-shoota** [comment]
@Sija Fixed. I think it would be great if the formatter would take care of this (at least the empty line before code blocks).

File: spec/std/string_spec.cr [hunk 3]
```diff
@@ -1440,54 +1440,58 @@ describe "String" do
...
-    "\#{".dump.should eq("\"\\\#{\"")
-    "á".dump.should eq("\"\\u00E1\"")
-    "\u{81}".dump.should eq("\"\\u0081\"")
+  it "#dump" do
+    "a".dump.should eq %("a")
+    "\\".dump.should eq %("\\\\")
+    "\"".dump.should eq %("\\\"")
+    "\b".dump.should eq %("\\b")
+    "\e".dump.should eq %("\\e")
+    "\f".dump.should eq %("\\f")
```
Comment (RX14) [inline spec/std/string_spec.cr]: `%q` and loose the escaping?

File: spec/std/string_spec.cr [hunk 4]
```diff
@@ -1440,54 +1440,58 @@ describe "String" do
...
-    "\#{".inspect.should eq("\"\\\#{\"")
-    "á".inspect.should eq("\"á\"")
-    "\u{81}".inspect.should eq("\"\\u0081\"")
+    "\u{1f48e}".dump.should eq %("\\u{1F48E}")
+  end
+
+  it "#dump_unquoted" do
+    "a".dump_unquoted.should eq %(a)
+    "\\".dump_unquoted.should eq %(\\\\)
+    "á".dump_unquoted.should eq %(\\u00E1)
```
Comment (RX14) [inline spec/std/string_spec.cr]: ditto

File: spec/std/string_spec.cr [hunk 5]
```diff
@@ -1440,54 +1440,58 @@ describe "String" do
...
+  it "#inspect" do
+    "a".inspect.should eq %("a")
+    "\\".inspect.should eq %("\\\\")
+    "\"".inspect.should eq %("\\\"")
+    "\b".inspect.should eq %("\\b")
+    "\e".inspect.should eq %("\\e")
+    "\f".inspect.should eq %("\\f")
+    "\n".inspect.should eq %("\\n")
+    "\r".inspect.should eq %("\\r")
+    "\t".inspect.should eq %("\\t")
```
Comment (RX14) [inline spec/std/string_spec.cr]: ditto

File: spec/std/string_spec.cr (hunk 3 above)
Comment (straight-shoota) [reply spec/std/string_spec.cr]: I don't mind changing it, but I'm not sure about the benefit. `%()` with added backslashes highlights that the resulting string is actually different from the original (`\f` vs. `\\f`).

File: src/string.cr [hunk 6]
```diff
@@ -3927,7 +3960,7 @@ class String
   end
 
   private def dump_char(char, error, io)
-    dump_or_inspect_char(char, error, io) do
+    dump_or_inspect_char char, error, io do
```
Comment (RX14) [inline src/string.cr]: I really dislike people making pointless changes which inflate diffs.

It's just as bad as people making pointless comments on PRs about style.

Comment (straight-shoota) [reply src/string.cr]: I wouldn't make that randomly. Just six lines above in an almost identical method defintion was literally the exact same line with different formatting. That felt too odd, really.

**Pull request merged**